code-samples/eventing/bookstore-sample-app/solution/solution.sh

```diff
@@ -95,58 +100,73 @@ read -p "🛑 Press ENTER to continue..."
 # Install the front end first
 echo ""
 echo "📦 Installing the Sample Bookstore Frontend..."
-cd frontend
+cd "${SCRIPT_DIR}/frontend"
 kubectl apply -f config
 
 # Wait for the frontend to be ready
 echo ""
 echo "⏳ Waiting for the frontend to be ready..."
 kubectl wait --for=condition=ready pod -l app=bookstore-frontend --timeout=300s
+echo "⏳ Waiting for the frontend load balancer to be ready... ensure kind cloud-provider-kind is active"
+kubectl wait --for=jsonpath='{.status.loadBalancer.ingress}' svc/bookstore-frontend-svc --timeout=300s
 echo "✅ Bookstore Frontend installed."
 
+FRONTEND_IP=$(kubectl get svc -o=jsonpath='{.status.loadBalancer.ingress[0].ip}' bookstore-frontend-svc)
+
+if [[ -z "$FRONTEND_IP" ]]; then
+    echo "Unable to get LoadBalancer IP for bookstore-frontend-svc"
+    exit 1
+fi
 
 # Prompt the user to check the frontend
 echo ""
-echo "✅ The frontend is now installed. Please visit http://localhost:3000 to view the bookstore frontend."
-echo "⚠️ If you cannot access the frontend, please open a new terminal and run 'kubectl port-forward svc/bookstore-frontend-svc 3000:3000' to forward the port to your localhost."
+echo "✅ The frontend is now installed. Please visit http://${FRONTEND_IP} to view the bookstore frontend."
 read -p '🛑 Can you see the front end page? If yes, press ENTER to continue...'
 
 # Install the node-server
 echo ""
 echo "📦 Installing the Sample Bookstore Backend (node-server)..."
-cd ../node-server
+cd "${SCRIPT_DIR}/node-server"
 kubectl apply -f config
 
 # Wait for the backend to be ready
 echo ""
 echo "⏳ Waiting for the backend to be ready..."
 kubectl wait --for=condition=ready pod -l app=node-server --timeout=300s
+echo "⏳ Waiting for the backend load balancer to be ready... ensure kind cloud-provider-kind is active"
+kubectl wait --for=jsonpath='{.status.loadBalancer.ingress}' svc/node-server-svc --timeout=300s
+
+NODE_SERVER_IP=$(kubectl get svc -o=jsonpath='{.status.loadBalancer.ingress[0].ip}' node-server-svc)
+
+if [[ -z "${NODE_SERVER_IP}" ]]; then
+    echo "Unable to get LoadBalancer IP for node-server-svc"
+    exit 1
+fi
 
 echo "✅ Bookstore Backend (node-server) installed."
 
 # Prompt the user to check the backend
 echo ""
-echo "✅ The node-server is now installed. Please visit http://localhost:8080 to view the bookstore node-server."
-echo "⚠️ If you cannot access it, please run 'kubectl port-forward svc/node-server-svc 8080:80' to forward the port to your localhost."
+echo "✅ The node-server is now installed. Please visit http://${NODE_SERVER_IP} to view the bookstore node-server."
 read -p '🛑 Can you see "Hello World!"? If yes, press ENTER to continue...'
 
 # Deploy the ML services
 echo ""
 echo "📦 Deploying the ML service: bad-word-filter..."
```

Comment:
`export DOCKER_HOST=unix://$HOME/.docker/run/docker.sock` is needed for me when using local KinD cluster. Otherwise, the following error prompt:
```
📦 Deploying the ML service: bad-word-filter...
Building function image
Error: cannot create docker client: docker/podman API not available
Possible causes:
  The docker/podman daemon is not running.
  The DOCKER_HOST environment variable is not set.

```

Reply:
`export DOCKER_HOST=unix://$HOME/.docker/run/docker.sock`

That's not universal so I don't think we should tell people to do that. A requirement beforehand should be a 'working docker'